app/ante/fee_checker_test.go

```diff
@@ -0,0 +1,305 @@
+
+	suite.clientCtx = client.Context{}.
+		WithTxConfig(suite.encCfg.TxConfig)
+
+	anteHandler, err := ante.NewAnteHandler(
+		ante.HandlerOptions{
+			AccountKeeper:   suite.accountKeeper,
+			BankKeeper:      suite.bankKeeper,
+			FeegrantKeeper:  suite.feeGrantKeeper,
+			SignModeHandler: suite.encCfg.TxConfig.SignModeHandler(),
+			SigGasConsumer:  ante.DefaultSigVerificationGasConsumer,
+		},
+	)
+
+	require.NoError(t, err)
+	suite.anteHandler = anteHandler
+
+	return suite
+}
+
+func (suite *AnteTestSuite) CreateTestAccounts(
+	t *testing.T,
+	numAccs uint64,
+) []TestAccount {
+	t.Helper()
+
+	var accounts []TestAccount
+
+	for i := uint64(0); i < numAccs; i++ {
+		priv, _, addr := testdata.KeyTestPubAddr()
+		acc := suite.accountKeeper.NewAccountWithAddress(suite.ctx, addr)
+		require.NoError(t, acc.SetAccountNumber(i))
+		suite.accountKeeper.SetAccount(suite.ctx, acc)
+		accounts = append(accounts, TestAccount{acc, priv})
+	}
+
+	return accounts
+}
+
+// TestDeductFees tests the full AnteHandler flow with the DeductFeeDecorator
+// using the default TxFeeChecker
+func TestDeductFees(t *testing.T) {
+	tests := []struct {
+		name             string
+		setupMocks       func(s *AnteTestSuite)
+		giveMinGasPrices sdk.DecCoins
+		giveFee          legacytx.StdFee
+		wantErr          string
+	}{
+		{
+			"insufficient funds for fee",
+			func(s *AnteTestSuite) {
+				s.bankKeeper.EXPECT().
+					SendCoinsFromAccountToModule(
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+					).
+					Return(sdkerrors.ErrInsufficientFunds)
+			},
+			sdk.NewDecCoins(sdk.NewDecCoinFromDec("ukava", sdk.MustNewDecFromStr("0.001"))),
+			legacytx.NewStdFee( //nolint:staticcheck // deprecated StdFee still in use
+				100000,
+				sdk.NewCoins(sdk.NewInt64Coin("ukava", 100)),
+			),
+			"insufficient funds: insufficient funds",
+		},
+		{
+			"valid fees",
+			func(s *AnteTestSuite) {
+				s.bankKeeper.EXPECT().
+					SendCoinsFromAccountToModule(
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+					).
+					Return(nil)
+			},
+			sdk.NewDecCoins(sdk.NewDecCoinFromDec("ukava", sdk.MustNewDecFromStr("0.001"))),
+			legacytx.NewStdFee( //nolint:staticcheck // deprecated StdFee still in use
+				100000,
+				sdk.NewCoins(sdk.NewInt64Coin("ukava", 100)), // gasPrice * gas
+			),
+			"",
+		},
+		{
+			"valid single fee, multiple min gas prices",
+			func(s *AnteTestSuite) {
+				s.bankKeeper.EXPECT().
+					SendCoinsFromAccountToModule(
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+					).
+					Return(nil)
+			},
+			sdk.NewDecCoins(
+				sdk.NewDecCoinFromDec("ukava", sdk.MustNewDecFromStr("0.001")),
+				sdk.NewDecCoinFromDec("usdt", sdk.MustNewDecFromStr("0.003")),
+			),
+			legacytx.NewStdFee( //nolint:staticcheck // deprecated StdFee still in use
+				100000,
+				sdk.NewCoins(sdk.NewInt64Coin("ukava", 100)),
+			),
+			"",
+		},
```

Comment:
```
{
			"valid single fee, multiple min gas prices",
			func(s *AnteTestSuite) {
				s.bankKeeper.EXPECT().
					SendCoinsFromAccountToModule(
						gomock.Any(),
						gomock.Any(),
						gomock.Any(),
						gomock.Any(),
					).
					Return(nil)
			},
			sdk.NewDecCoins(
				sdk.NewDecCoinFromDec("ukava", sdk.MustNewDecFromStr("0.001")),
				sdk.NewDecCoinFromDec("akava", sdk.MustNewDecFromStr("10000000000")),
				sdk.NewDecCoinFromDec("usdt", sdk.MustNewDecFromStr("0.003")),
			),
			legacytx.NewStdFee( //nolint:staticcheck // deprecated StdFee still in use
				100000,
				sdk.NewCoins(sdk.NewInt64Coin("usdt", 300)),
			),
			"",
		},
```


Nit: maybe update this test to mimic mainnet conditions more closely and use USDT as fee coin? Tested all the variations (akava for fees, usdt for fees, etc) and it's working as expected.